### ISSUE TITLE
Implement Fixed Header Javascript Solution

### ISSUE THREAD

**Issue by dbjpanda** [tscanlin/tocbot#170]
I am creating this issue again as it is one of the most shout out issues. A javascript solution has not been added in the plugin because it is managed to replace with a simple CSS fix. 

But if you go through this StackOverflow [Question](https://stackoverflow.com/questions/10732690/offsetting-an-html-anchor-to-adjust-for-fixed-header), it has been seen in many cases, fixing by CSS results in potential issue in presentation layer.

Apart from it, In most cases, `margin-top = - (padding-top)` doesn't work, you have to manually adjust both the properties separately which is not as much as flexible if you are using this TOC Bot library behind another distributed software, in my case it is [TOC Pack](https://dbjpanda.me/apps/tocpack). 

However, as the plugin author @tscanlin commented [here](https://github.com/tscanlin/tocbot/pull/158#issuecomment-567734956) that he is now open to accept a PR on this, I am opening the issue again. I will try to work on it and may send a PR soon. In case anyone interested in the meantime can contribute as well. @tscanlin It would be better if you can add a label as **Help Needed**  :)

**Comment by tscanlin** [comment]
Can you link me to an example where the CSS solution doesn't work?

The main problem for any JS based solution would be page load. I'd expect there may be a flash when the scroll position is changed from where it would be natively to the JS provided offset to account for a fixed header. With the CSS solution the native scroll position is changed to account for the header.

Please let me know if anyone has time or ideas for how to implement this.

**Comment by dbjpanda** [comment]
@tscanlin I was actually trying to use css fix recently on a client's site https://www.photo8.co.uk . I needed a lot of adjustment in padding and yet the heading was not positioned properly. So shifted to another JavaScript solution. You may check that.

**Comment by tscanlin** [comment]
There's some other CSS that may be interfering with the pure CSS solution
```
.elementor-heading-title {
    padding-top: 163px!important;
    margin-top: -106px!important;
}
```
Fix that and the CSS way should work.

**Comment by dbjpanda** [comment]
Ahh. Actually, now I don't have any access to that site. It was added by me initially when the TOC Bot library had been used. The client was not happy with this solution because of the heading tags are not positioned properly. So he shifted to another solution and probably have forgotten to remove these CSS. :)

**Comment by tscanlin** [comment]
Can you link me to a reproduction of this issue @dbjpanda ?

**Comment by ayZagen** [comment]
CSS solution will make elements above applied element non clickable.

**Comment by tscanlin** [comment]
Fixed in #213
